### PR TITLE
upgrade templates to use gradle dsl in android apps

### PR DESCRIPTION
**Pull request by AbdeMohlbi**
towards [#151166](https://github.com/flutter/flutter/issues/151166)
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: packages/flutter_tools/test/src/android_common.dart [hunk 1]
```diff
@@ -43,30 +43,34 @@ class FakeAndroidBuilder implements AndroidBuilder {
...
+          {required FlutterProject project}) async =>
+      const <String>[];
 
   @override
   Future<String> outputsAppLinkSettings(
     String buildVariant, {
     required FlutterProject project,
-  }) async => '/';
+  }) async =>
+      '/';
```
Comment (nate-thegrate) [inline packages/flutter_tools/test/src/android_common.dart:55]: These auto-formatter changes are unfortunately a readability downgrade from the original. We could either revert or replace with something better:

```dart
@override
  Future<List<String>> getBuildVariants({required FlutterProject project}) {
    return Future.value(const <String>[]);
  }

  @override
  Future<String> outputsAppLinkSettings(
    String buildVariant, {
    required FlutterProject project,
  }) {
    return Future.value('/');
  }
```

File: packages/flutter_tools/test/src/android_common.dart [hunk 2]
```diff
@@ -43,30 +43,34 @@ class FakeAndroidBuilder implements AndroidBuilder {
...
+        );
 
   final file_system.Directory directoryOverride;
 
   @override
   FlutterProject fromDirectory(Directory _) {
     projects.clear();
-    return super.fromDirectory(directoryOverride.childDirectory('flutter_project'));
+    return super
+        .fromDirectory(directoryOverride.childDirectory('flutter_project'));
```
Comment (nate-thegrate) [inline packages/flutter_tools/test/src/android_common.dart:73]: Let's either revert or replace with:

```dart
    return super.fromDirectory(
      directoryOverride.childDirectory('flutter_project'),
    );
```

File: packages/flutter_tools/test/src/android_common.dart [hunk 3]
```diff
@@ -156,33 +161,35 @@ Future<ProcessResult> buildFlutterApkWithSpecifiedDependencyVersions({
...
+      gradleWrapperPropertiesFileContent.replaceFirst(
     gradleReplacementString,
     versions.gradleVersion,
   );
   await gradleWrapperProperties.writeAsString(propertyContent, flush: true);
 
-  final File gradleSettings = File(fileSystem.path.join(
-      app.path, 'android', 'settings.gradle'));
+  final File gradleSettings =
+      File(fileSystem.path.join(app.path, 'android', 'settings.gradle.kts'));
```
Comment (nate-thegrate) [inline packages/flutter_tools/test/src/android_common.dart:187]:
```suggestion
  final File gradleSettings = File(
    fileSystem.path.join(app.path, 'android', 'settings.gradle.kts'),
  );
```

File: packages/flutter_tools/test/src/android_common.dart [hunk 4]
```diff
@@ -156,33 +161,35 @@ Future<ProcessResult> buildFlutterApkWithSpecifiedDependencyVersions({
...
         .replaceFirst(flutterCompileSdkString, versions.compileSdkVersion!);
     appGradleBuild.writeAsStringSync(appBuildContent);
   }
 
   // Modify gradle version to passed in version.
   final File gradleWrapperProperties = File(fileSystem.path.join(
       app.path, 'android', 'gradle', 'wrapper', 'gradle-wrapper.properties'));
-  final String propertyContent = gradleWrapperPropertiesFileContent.replaceFirst(
+  final String propertyContent =
+      gradleWrapperPropertiesFileContent.replaceFirst(
```
Comment (nate-thegrate) [inline packages/flutter_tools/test/src/android_common.dart:180]: Let's revert this one.

File: packages/flutter_tools/test/src/android_common.dart [hunk 5]
```diff
@@ -156,33 +161,35 @@ Future<ProcessResult> buildFlutterApkWithSpecifiedDependencyVersions({
...
 
   if (versions.compileSdkVersion != null) {
-    final File appGradleBuild = File(fileSystem.path.join(
-        app.path, 'android', 'app', 'build.gradle'));
-    final String appBuildContent = appGradleBuild.readAsStringSync()
+    final File appGradleBuild = File(
+        fileSystem.path.join(app.path, 'android', 'app', 'build.gradle.kts'));
+    final String appBuildContent = appGradleBuild
+        .readAsStringSync()
         .replaceFirst(flutterCompileSdkString, versions.compileSdkVersion!);
```
Comment (nate-thegrate) [inline packages/flutter_tools/test/src/android_common.dart:172]: I think `appBuildContent` looks better now than it did before 👍

`appGradelBuild` should probably be changed to:

```dart
    final File appGradleBuild = File(
      fileSystem.path.join(app.path, 'android', 'app', 'build.gradle.kts'),
    );
```

**Review by nate-thegrate — COMMENTED** [review]
Unfortunately I don't have enough experience with `.tmpl` files to effectively review them.

I will say that the "++" title should be changed, since that's what the `autosubmit` bot will use to label the commit when it merges the PR.

**Comment by andrewkolos** [comment]
Could you update this PR with a title and description?

**Comment by AbdeMohlbi** [comment]
> Unfortunately I don't have enough experience with `.tmpl` files to effectively review them.
> 
> I will say that the "++" title should be changed, since that's what the `autosubmit` bot will use to label the commit when it merges the PR.

I didn't intend to request a review for this pr for the moment a lot of work is missing like updating the template files of plugins... more tests

**Comment by AbdeMohlbi** [comment]
> Please add a title and description to this PR.

I did this on purpose to avoid getting a review, i haven't finished the work yet

File: packages/flutter_tools/test/src/android_common.dart (hunk 1 above)
Comment (AbdeMohlbi) [reply packages/flutter_tools/test/src/android_common.dart:55]: Hmmm i'll see to it then

File: packages/flutter_tools/test/src/android_common.dart (hunk 2 above)
Comment (AbdeMohlbi) [reply packages/flutter_tools/test/src/android_common.dart:73]: Ok

**Comment by AbdeMohlbi** [comment]
> > i haven't finished the work yet
> 
> There's a really handy button you can use:
> 
> draft button

light theme 😮

**Comment by nate-thegrate** [comment]
> light theme 😮

I got used to GitHub light theme, now it's too late to go back 🥲

File: packages/flutter_tools/test/src/android_common.dart (hunk 1 above)
Comment (AbdeMohlbi) [reply packages/flutter_tools/test/src/android_common.dart:55]: i'm getting Missing type annotation. now ?

Comment (AbdeMohlbi) [reply packages/flutter_tools/test/src/android_common.dart]: @nate-thegrate what do u think ?

**Comment by AbdeMohlbi** [comment]
@bartekpacia i guess this could help for now , not sure about the other tests tho

**Comment by bartekpacia** [comment]
I don't understand why you're duplicating the same work I'm doing in https://github.com/flutter/flutter/pull/154061

**Comment by nate-thegrate** [comment]
I hadn't noticed till now, but based on the `help_with_154061` branch name, it's clear that the goal of this PR is to try to help @bartekpacia with the efforts being made in #154061.

Setting the target branch as `bartekpacia:gradle_kotlin_by_default` would have been more pertinent than `flutter:master`. Even then, this is not something I would recommend, per [Brooks's Law](https://en.wikipedia.org/wiki/Brooks%27s_law):

> What one programmer can do in one month, two programmers can do in two months.

<br>

I'm going to close this one. In the future, my recommendation would be to avoid work that overlaps with another person's contribution—I've learned the hard way that even with good intentions, [things can still break](https://github.com/flutter/flutter/pull/151144).

**Pull request closed without merging**